source/slang/slang-check-overload.cpp

```diff
@@ -1218,6 +1229,11 @@ namespace Slang
             auto overloadRankDiff = getOverloadRank(right->item.declRef) - getOverloadRank(left->item.declRef);
             if (overloadRankDiff)
                 return overloadRankDiff;
+
+            // `export` function is more flavored than `extern` function. But other modifiers are not considered.
+            auto externExportDiff = getExportRank(left->item.declRef, right->item.declRef);
```

Comment:
This check should be before overload rank. We use overload rank as the last resort to disambiguate and it is really just a hack and should not take precedence over anything.